### PR TITLE
Add test coverage for ComboBox

### PR DESCRIPTION
**Pull request by Nora-Zhou01**
Related https://github.com/dotnet/winforms/issues/10453
## Proposed changes

- Add unit tests for ComboBox to test its methods: GetItemHeight, ResetText, SelectAll
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11533)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 1]
```diff
@@ -511,6 +511,102 @@ public void ComboBox_DropDownStyle_Set_GetReturnsExpected(ComboBoxStyle value)
...
+        using ComboBox control = new();
+        control.DrawMode = DrawMode.OwnerDrawVariable;
+        control.CreateControl();
+        control.Invoking(c => c.GetItemHeight(0))
+               .Should().Throw<ArgumentOutOfRangeException>();
+    }
+
+    [WinFormsTheory]
+    [InlineData(false)]
+    [InlineData(true)]
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:573]: Please using the [BoolData] in here if needed

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 2]
```diff
@@ -511,6 +511,102 @@ public void ComboBox_DropDownStyle_Set_GetReturnsExpected(ComboBoxStyle value)
         Assert.False(control.IsHandleCreated);
     }
 
+    [WinFormsTheory]
+    [InlineData(DrawMode.Normal, 0, 15)]
+    [InlineData(DrawMode.OwnerDrawFixed, 0, 15)]
+    [InlineData(DrawMode.OwnerDrawVariable, 0, 15)]
+    public void ComboBox_GetItemHeight_Invoke_ReturnsExpected(DrawMode drawMode, int index, int expected)
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:518]: It seems like the only thing changing here is the `DrawMode`, so let's inline `index` and `expected` value in the test code and remove these parameters

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 3]
```diff
@@ -511,6 +511,109 @@ public void ComboBox_DropDownStyle_Set_GetReturnsExpected(ComboBoxStyle value)
...
     }
 
+    [WinFormsTheory]
+    [InlineData(DrawMode.Normal)]
+    [InlineData(DrawMode.OwnerDrawFixed)]
+    [InlineData(DrawMode.OwnerDrawVariable)]
+    public void ComboBox_GetItemHeight_Invoke_ReturnsExpected(DrawMode drawMode)
+    {
+        const int index = 0;
+        const int expected = 15;
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:521]:
```suggestion
        int index = 0;
        int expected = 15;
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 4]
```diff
@@ -511,6 +511,109 @@ public void ComboBox_DropDownStyle_Set_GetReturnsExpected(ComboBoxStyle value)
...
+    {
+        using ComboBox control = new();
+        control.DrawMode = DrawMode.OwnerDrawVariable;
+        control.CreateControl();
+        control.Invoking(c => c.GetItemHeight(0))
+               .Should().Throw<ArgumentOutOfRangeException>();
+    }
+
+    [WinFormsTheory]
+    [MemberData(nameof(BoolData))]
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:574]: There is already an existing bool data that can be used
```suggestion
    [BoolData]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs [hunk 5]
```diff
@@ -511,6 +511,109 @@ public void ComboBox_DropDownStyle_Set_GetReturnsExpected(ComboBoxStyle value)
...
+        control.Text = "Some text";
+        control.ResetText();
+        Assert.Equal(string.Empty, control.Text);
+    }
+
+    public static IEnumerable<object[]> BoolData()
+    {
+        yield return new object[] { false };
+        yield return new object[] { true };
+    }
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ComboBoxTests.cs:592]: Please remove this

**Review by lonitra — COMMENTED** [review]
Few more comments, otherwise LGTM

**Comment by Nora-Zhou01** [comment]
This branch has conflicts

**Pull request closed without merging**